src/posit/connect/resources.py

```diff
@@ -43,3 +46,254 @@ def update(self, *args, **kwargs):
+        The results are cached after the first request and reused for subsequent access unless reloaded.
+
+        Returns
+        -------
+        List[T_co]
+            A list of items of type `T_co` representing the fetched data.
+        """
+        if self._cache:
+            return self._cache
+
+        response = self._ctx.session.get(self._endpoint)
+        results = response.json()
+        self._cache = [self._cls(self._ctx, self._parent, **result) for result in results]
+        return self._cache
+
+    @overload
+    def __getitem__(self, index: int) -> T_co: ...
+
+    @overload
+    def __getitem__(self, index: slice) -> Sequence[T_co]: ...
+
+    def __getitem__(self, index):
+        return self._data[index]
+
+    def __len__(self) -> int:
+        return len(self._data)
+
+    def __str__(self) -> str:
+        return str(self._data)
+
+    def __repr__(self) -> str:
+        return repr(self._data)
+
+    def reload(self) -> "ActiveSequence":
+        """
+        Clear the cache and reload the data from the API on the next access.
+
+        Returns
+        -------
+        ActiveSequence
+            The current instance with cleared cache, ready to reload data on next access.
+        """
+        self._cache = None
+        return self
+
+
+class ActiveFinderMethods(ActiveSequence[T_co], ABC, Generic[T_co]):
+    """
+    Finder methods.
+
+    Provides various finder methods for locating records in any endpoint supporting HTTP GET requests.
+
+    Attributes
+    ----------
+    _uid : str
+        The default field name used to uniquely identify records. Defaults to 'guid'.
+
+    Methods
+    -------
+    find(uid) -> T_co
+        Finds and returns a record by its unique identifier (`uid`). If a cached result exists, it searches within the cache;
+        otherwise, it makes a GET request to retrieve the data from the endpoint.
+
+    find_by(**conditions: Any) -> Optional[T_co]
+        Finds the first record that matches the provided conditions. If no record is found, returns None.
+    """
+
+    _uid: str = "guid"
+    """The default field name used to uniquely identify records. Defaults to 'guid'."""
+
+    def find(self, uid) -> T_co:
+        """
+        Find a record by its unique identifier.
+
+        Fetches a record either by searching the cache or by making a GET request to the endpoint.
+
+        Parameters
+        ----------
+        uid : Any
+            The unique identifier of the record.
+
+        Returns
+        -------
+        T_co
+
+        Raises
+        ------
+        ValueError
+            If no record is found.
+        """
+        if self._cache:
+            conditions = {self._uid: uid}
+            result = self.find_by(**conditions)
+        else:
+            endpoint = self._endpoint + uid
+            response = self._ctx.session.get(endpoint)
+            result = response.json()
+            result = self._cls(self._ctx, self._parent, **result)
```

Comment:
Could use talking through how the operators work here &… well, I feel fuzzy on a few questions about how this works when you're finding by something that isn't the key. Does it get all the jobs from the endpoint and find within those? I know I'm misunderstanding some part of this relationship.

Reply:
> Does it get all the jobs from the endpoint and find within those?

That's correct.